jt-scale2/src/main/java/it/geosolutions/jaiext/scale/Scale2NearestOpImage.java

```diff
@@ -2112,4 +2122,12 @@ private void doubleLoop(RasterAccessor src, Rectangle dstRect, RasterAccessor ds
             }
         }
     }
+
+    private final static int checkPos(int pos, int overflowPos) {
+        // In some rare corner-cases the scaling computation  approximations
+        // can result into having posX + posY equal to the databuffer length.
+        // Let's round it back to the last valid pixel
+        return  (!DATABUFFER_WORKAROUND || pos != overflowPos) ? pos : pos - 1;
```

Comment:
Hmm... is it really the same thing as the databuffer workaround though? What about:

```suggestion
        return  (pos < overflowPos) ? pos : overflowPos - 1;
```

Reply:
Well. It's still related to databuffer containing 1 pixel less than expected.
I intentionally didn't generalize to cases like pos >= overflowPos since I suspect it will hide some higher level issues in the chain processing. 
I mean... if we get overflowPos = 1000 and pos = 1800 there is something wrong somewhere in the chain or in the layout and setting 999 will probably delay the problem in some other piece of code or show bad rendering.
Does it make sense?

Reply:
It does thanks